lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -246,106 +248,207 @@ def test_clear_tools_test72(monkeypatch, agent_config, pbench_run, pbench_cfg):
 
     another_group = pbench_run / "tools-v1-another"
     another_group.mkdir(parents=True)
-    fubar_another_host = another_group / "fubar7.example.com"
+    fubar_another_host = another_group / "fubar.example.com"
     fubar_another_host.mkdir(parents=True, exist_ok=True)
     pidstat_another_tool = fubar_another_host / "pidstat"
     pidstat_another_tool.touch()
 
-    command = ["pbench-clear-tools", "--group=another,wrong"]
+    command = ["pbench-clear-tools", "--groups=another,wrong"]
     out, err, exitcode = pytest.helpers.capture(command)
     assert b"" == out
     assert (
-        b'Removed "pidstat" from host "fubar7.example.com" in tools group '
+        b'Removed "pidstat" from host "fubar.example.com" in tools group '
         b'"another"' in err
     )
     assert b'pbench-clear-tools: invalid --group option "wrong"' in err
     assert another_group.exists() is False
 
 
-def test_clear_tools_test73(monkeypatch, agent_config, pbench_run, pbench_cfg):
+@pytest.mark.parametrize(
+    "groups", ["default", "default,custom1", "default,custom1,custom2"]
+)
+@pytest.mark.parametrize(
+    "remotes",
+    ["remote1", "remote1,remote2", "remote1,remote2,remote3", "remote2,remote4"],
+)
+@pytest.mark.parametrize(
+    "tools", ["pidstat", "pidstat,mpstat", "pidstat,mpstat,iostat", "mpstat," "vmstat"]
+)
+def test_clear_tools_test73(
+    monkeypatch, tmp_path, agent_config, pbench_run, pbench_cfg, groups, remotes, tools
+):
     """ Attempt to clear the tools, by name, separated by comma"""
     monkeypatch.setenv("_PBENCH_AGENT_CONFIG", str(pbench_cfg))
-    default_group = pbench_run / "tools-v1-default"
-    default_group.mkdir(parents=True)
-    fubar6_host = default_group / "fubar6.example.com"
-    fubar6_host.mkdir(parents=True, exist_ok=True)
-    pidstat6_tool = fubar6_host / "pidstat"
-    pidstat6_tool.touch()
-    mpstat6_tool = fubar6_host / "mpstat"
-    mpstat6_tool.touch()
-    iostat6_tool = fubar6_host / "iostat"
-    iostat6_tool.touch()
+    tools_existence = []
+    for group in groups.split(","):
+        custom_group = pbench_run / f"tools-v1-{group}"
+        custom_group.mkdir(parents=True)
+        for remote in remotes.split(","):
+            remote_host = custom_group / f"{remote}.example.com"
+            remote_host.mkdir(parents=True, exist_ok=True)
+            for tool in tools.split(","):
+                tool_file = remote_host / tool
+                tool_file.touch()
+                tools_existence.append(tool_file)
 
     command = ["pbench-clear-tools", "--name=pidstat,mpstat"]
     out, err, exitcode = pytest.helpers.capture(command)
     assert b"" == out
     assert b"" in err
-    assert pidstat6_tool.exists() is False
-    assert mpstat6_tool.exists() is False
-    assert iostat6_tool.exists() is True
+    for tool in tools_existence:
+        tool_str = str(tool).replace(str(tmp_path), "")
```

Comment:
Unfortunately, what happens is because I am using the parameterization, pytest creates this weird tmp directory structure for nested combinations 

```
lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py::test_clear_tools_test86[mpstat,vmstat-remote1.example.com,remote4.example.com-default,group4] 
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat0/var/lib/pbench-agent/tools-v1-default/remote1.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat0/var/lib/pbench-agent/tools-v1-default/remote2.example.com/mpstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat0/var/lib/pbench-agent/tools-v1-default/remote3.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat0/var/lib/pbench-agent/tools-v1-default/remote3.example.com/turbostat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat0/var/lib/pbench-agent/tools-v1-group1/remote1.example.com/mpstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat0/var/lib/pbench-agent/tools-v1-group3/remote3.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat0/var/lib/pbench-agent/tools-v1-group3/remote3.example.com/mpstat
PASSED
lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py::test_clear_tools_test86[mpstat,vmstat-remote1.example.com,remote4.example.com-group1,group4] 
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat1/var/lib/pbench-agent/tools-v1-default/remote1.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat1/var/lib/pbench-agent/tools-v1-default/remote2.example.com/mpstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat1/var/lib/pbench-agent/tools-v1-default/remote3.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat1/var/lib/pbench-agent/tools-v1-default/remote3.example.com/turbostat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat1/var/lib/pbench-agent/tools-v1-group1/remote1.example.com/mpstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat1/var/lib/pbench-agent/tools-v1-group3/remote3.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat1/var/lib/pbench-agent/tools-v1-group3/remote3.example.com/mpstat
PASSED
lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py::test_clear_tools_test86[mpstat,vmstat-remote4.example.com,remote5.example.com-default,group4] 
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat2/var/lib/pbench-agent/tools-v1-default/remote1.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat2/var/lib/pbench-agent/tools-v1-default/remote2.example.com/mpstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat2/var/lib/pbench-agent/tools-v1-default/remote3.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat2/var/lib/pbench-agent/tools-v1-default/remote3.example.com/turbostat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat2/var/lib/pbench-agent/tools-v1-group1/remote1.example.com/mpstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat2/var/lib/pbench-agent/tools-v1-group3/remote3.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat2/var/lib/pbench-agent/tools-v1-group3/remote3.example.com/mpstat
PASSED
lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py::test_clear_tools_test86[mpstat,vmstat-remote4.example.com,remote5.example.com-group1,group4] 
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat3/var/lib/pbench-agent/tools-v1-default/remote1.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat3/var/lib/pbench-agent/tools-v1-default/remote2.example.com/mpstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat3/var/lib/pbench-agent/tools-v1-default/remote3.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat3/var/lib/pbench-agent/tools-v1-default/remote3.example.com/turbostat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat3/var/lib/pbench-agent/tools-v1-group1/remote1.example.com/mpstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat3/var/lib/pbench-agent/tools-v1-group3/remote3.example.com/pidstat
/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat3/var/lib/pbench-agent/tools-v1-group3/remote3.example.com/mpstat
PASSED

```
and so if I want to check if the `mpstat` exists in the path its will return true in every situation as you can see the directory contains `mpstat` in every combination. Therefore I decided to use the `tmp_path` to remove the prefix, in this case `/tmp/pytest-of-nikhilpalaskar/pytest-209/test_clear_tools_test86_mpstat3` and what we are left with is `/var/lib/pbench-agent/tools-v1-default/remote2.example.com/mpstat` which is a right directory structure for our search.

Reply:
I wasn't suggesting that the code shouldn't remove the prefix; rather, I was suggesting that it should remove the `pbench_run` prefix instead of the `tmp_path` prefix.  I believe that that would reduce each path to just `tools-v1-<group>/<remote>/<tool>` which is even less likely to encounter interference (and it reduces the dependencies of the test function, since the `tmp_path` fixture is no longer (directly) referenced).